### PR TITLE
 doc: zephyr_domain: Cross reference relevant API in code samples

### LINKED ISSUE #77149: Automatically include references to a sample's relevant APIs
It would be nice that, when a sample has `:relevant-api:` specified in its `README.rst`, the listed APIs are automatically linked to in the generated documentation of the sample.

Maybe as an automatically added section (`API reference`/`Relevent API(s)`) at the bottom of the page?

### PR DESCRIPTION
**Pull request by kartben**
Auto-add references to doxygen at the bottom of code samples' README
Fixes #77149.

Example of the new output: https://builds.zephyrproject.io/zephyr/pr/77159/docs/samples/synchronization/README.html

### CODE REVIEW

File: doc/_extensions/zephyr/domain.py [hunk 1]
```diff
@@ -111,6 +114,31 @@ def convert_node(self, node):
             for sibling in siblings_to_move:
                 parent.remove(sibling)
 
+            # Add a "See Also" section at the end with links to relevant APIs
+            if node["relevant-api"]:
+                see_also_section = nodes.section(ids=[f"{node['id']}-see-also"])
+                see_also_section += nodes.title(text="See Also")
```
Comment (tomi-font) [inline doc/_extensions/zephyr/domain.py:120]: How about capitalizing only the first word? `See also`

File: doc/_extensions/zephyr/domain.py [hunk 2]
```diff
@@ -111,6 +114,31 @@ def convert_node(self, node):
             for sibling in siblings_to_move:
                 parent.remove(sibling)
 
+            # Add a "See Also" section at the end with links to relevant APIs
+            if node["relevant-api"]:
+                see_also_section = nodes.section(ids=[f"{node['id']}-see-also"])
```
Comment (tomi-font) [inline doc/_extensions/zephyr/domain.py:119]: Could the anchor name be just `see-also`?

**Review by tomi-font — COMMENTED** [review]
Thank you for this!

**Review by gmarull — DISMISSED** [review]

**Review by awojasinski — DISMISSED** [review]

**Comment by kartben** [comment]
addressed @tomi-font's comments.

**Review by gmarull — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**